### PR TITLE
ike: set event for multiple server proposals - v2

### PR DESCRIPTION
**Pull request by fhonza**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4433

Describe changes ( from #6061 ):
- set event only once
- added documentation for log output in case of multiple server transforms

### CODE REVIEW

File: rust/src/ike/detect.rs [hunk 1]
```diff
@@ -151,17 +151,16 @@ pub extern "C" fn rs_ike_state_get_sa_attribute(
...
-                    break;
-                }
-                for attr in server_transform {
-                    if attr.attribute_type.to_string() == sa {
-                        if let Some(numeric_value) = attr.numeric_value {
-                            ret_val = numeric_value;
-                            ret_code = 1;
-                            break 'outer1;
+            if tx.hdr.ikev1_transforms.len() >= 1 {
+                // there should be only one chosen server_transform, check event
```
Comment (catenacyber) [inline rust/src/ike/detect.rs:155]: Maybe this deserves a comment in the documentation about the keyword...

File: rust/src/ike/ikev1.rs [hunk 2]
```diff
@@ -130,6 +130,11 @@ pub fn handle_ikev1(
                         &tx.hdr.ikev1_transforms,
                     );
                 } else {
+                    if state.ikev1_container.server.transforms.len() <= 1 && state.ikev1_container.server.transforms.len() + tx.hdr.ikev1_transforms.len() > 1 {
```
Comment (catenacyber) [inline rust/src/ike/ikev1.rs:133]: Can we rustfmt this to get a shorter line ?

**Review by catenacyber — APPROVED** [review]
Looks good to me.
I added 2 questions that may be improved...

**Comment by fhonza** [comment]
> Looks good to me.
> I added 2 questions that may be improved...

Have added both of your points, hope the force push was ok for this :)

**Review by catenacyber — APPROVED** [review]
Looks good to me, thanks

**Comment by victorjulien** [comment]
Merged in #6101, thanks!

**Pull request closed without merging**